### DISCUSSION TITLE
Why re-invent the wheel?

### DISCUSSION THREAD

**Discussion by ArjixWasTaken** [microsoft/sudo#26] [General]
There has been an excellent project named [gsudo](https://github.com/gerardog/gsudo) and it has an alias for `sudo` as well.
It works just the way one would expect it to, and it has existed since 2019 (~5 years as of writing).

It is available on winget, scoop and chocolatey.
I'd assume it would make more sense to either sponsor that guy or to fork their project.
Why are you making your own from scratch?

> Sudo is available for Windows 11 builds 26045 and later. If you're on an Insiders build with sudo, you can enable it in the Windows Settings app, on the "Developer Features" page.

Are you planning to have your version of sudo enabled by default once it is mature?
~~Would it conflict with gsudo?~~

Update: It does indeed conflict, even when disabled...
Solution: Edit your system environment for PATH to have the gsudo directory before `C:\Windows\System32`

**Comment by mwallner** [comment]
Déjà-vu ... it feels like another :moyai: "oh we need to have this up in our toolbelt - let's create it"  
=> and 100% forget about server-OS and LTS releases. :slowclap:

**Comment by Enivex** [comment]
They specifically mention gsudo in their blog post https://devblogs.microsoft.com/commandline/introducing-sudo-for-windows/

**Reply by ArjixWasTaken** [reply]
Thanks for the link!
Although upon reading it, they merely say that if microsoft's sudo is lacking, they recommend using gsudo...

Which makes me further question their actions.

**Comment by ZeyadMoustafaKamal** [comment]
Sometimes I feel like using Windows and any related product/tool is a waste of time.

I still remember when I was using Winget for the first time. I just wanted to install Python just to try it and suddenly I got a GUI with these "next next next" to install Python. which makes me ask are you sure that this is a package manager?"

I think that the only tool related to Windows that I thank Microsoft for is WSL.

**Comment by NamelessGO** [comment]
Comparison between gsudo and Microsoft sudo

https://gerardog.github.io/gsudo/docs/gsudo-vs-sudo

**Reply by mwallner** [reply]
I didn't realize MSFT's sudo doesn't even return the exit code of the command called... that's a BIG red flag for me! (guess that'll be fixed soon though if they want people to actually being able to use it)

**Reply by zadjii-msft** [reply]
> doesn't even return the exit code of the command called

It sure should! Seems like that regressed at some point before 0.1.5 went out to Insiders, then got re-fixed (since I can't repro that on my main branch anymore). 

Needless to say, I suspect that'll be fixed in the next version of sudo that lands in Insiders (in ~2-3 weeks). I'll make a note in #4

**Comment by ivorybishop** [comment]
If you are going to reinvent the wheel, why not make it work with all the other wheels out there?

It's not that hard to make it so that everyone adopting your tool doesn't have to modify scripts to make it work.
